### PR TITLE
Update release notes for RC2 release of Providers for May 2022

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by josh-fell — APPROVED** [review]

**Comment by potiuk** [comment]
I tihnk this is the last one: https://github.com/apache/airflow/pull/24330 I will rebase this one and update and Can release RC2 after that.

**Comment by potiuk** [comment]
Hey @josh-fell - this one contains pre-commit that fixes #24331 in the new provider's release :). I will use the code - slightly modified - to also fix historical pages later, when we see that one working.

File: docs/apache-airflow-providers-amazon/example-dags.rst [hunk 1]
```diff
@@ -20,4 +20,4 @@ Example DAGs
 
 You can learn how to use Amazon AWS integrations by analyzing the source code of the example DAGs:
 
-* `Amazon AWS <https://github.com/apache/airflow/tree/main/airflow/providers/amazon/aws/example_dags>`__
+* `Amazon AWS <https://github.com/apache/airflow/tree/providers-amazon/4.0.0/tests/system/providers/amazon/aws>`__
```
Comment (josh-fell) [inline docs/apache-airflow-providers-amazon/example-dags.rst:23]: For the providers that are still in progress for their transition from example DAGs to system tests (e.g. Amazon, Google), should we default to pointing to system tests even if there are still example DAGs in the original directory (and in the case of Amazon, the vast majority haven't been migrated)? Maybe we check if the example DAG path exists first in the pre-commit check instead?

Comment (potiuk) [reply docs/apache-airflow-providers-amazon/example-dags.rst:23]: Well it's wrong both ways in fact ... But I could modify it slightly to add BOTH links in this case :) Hold my beer.

Comment (potiuk) [reply docs/apache-airflow-providers-amazon/example-dags.rst:23]: Should be fine now :) . We should have legacy links now for both google and amazon (those are the only ones which  are in-progress.. And the legacy link will disappear when completed.

While doing so, I also fixed an inconsistency in google examples - the "cloud" examples weren't in "cloud" subfolder in tests/systems/. This is now fixed and verified by pre-commit too, so it was worth the effort :)

File: docs/apache-airflow-providers-google/example-dags.rst [hunk 2]
```diff
@@ -20,9 +20,10 @@ Example DAGs
...
 
-* `Google Ads <https://github.com/apache/airflow/tree/main/airflow/providers/google/ads/example_dags>`__
-* `Google Cloud <https://github.com/apache/airflow/tree/main/airflow/providers/google/cloud/example_dags>`__
-* `Google Firebase <https://github.com/apache/airflow/tree/main/airflow/providers/google/firebase/example_dags>`__
-* `Google Marketing Platform <https://github.com/apache/airflow/tree/main/airflow/providers/google/marketing_platform/example_dags>`__
-* `Google Workplace <https://github.com/apache/airflow/tree/main/airflow/providers/google/suite/example_dags>`__ (formerly Google Suite)
-* `Google LevelDB <https://github.com/apache/airflow/tree/main/airflow/providers/google/leveldb/example_dags>`__
+* `Google Ads <https://github.com/apache/airflow/tree/providers-google/8.0.0/airflow/providers/google/ads/example_dags>`__
+* `Google Cloud <https://github.com/apache/airflow/tree/providers-google/8.0.0/airflow/providers/google/cloud/example_dags>`__
+* `Google Cloud (legacy) <https://github.com/apache/airflow/tree/providers-google/8.0.0/tests/system/providers/google/cloud>`__
```
Comment (josh-fell) [inline docs/apache-airflow-providers-google/example-dags.rst]:
```suggestion
* `Google Cloud (legacy examples) <https://github.com/apache/airflow/tree/providers-google/8.0.0/airflow/providers/google/cloud/example_dags>`__
* `Google Cloud <https://github.com/apache/airflow/tree/providers-google/8.0.0/tests/system/providers/google/cloud>`__
```
To be consistent with Amazon with `example_dags/` being legacy. I don't feel strongly on "(legacy)" vs "(legacy examples)" verbiage though.

**Review by josh-fell — APPROVED** [review]

**Pull request merged**